### PR TITLE
[Logs agent] Disable HTTP/2 when using a proxy

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Out of an abundance of caution, disable HTTP/2 negotiation for the logs agent when a proxy is configured. 
This will be removed in a future agent version. 

### Motivation

Logs agent transport improvements. 

### Describe how you validated your changes

QA covered by unit test (as it spins up a real server/client)

### CODE REVIEW

File: pkg/logs/client/http/destination_test.go [hunk 1]
```diff
@@ -494,3 +494,34 @@ func TestTransportProtocol_HTTP1FallBack(t *testing.T) {
...
 }
+
+func TestTransportProtocol_HTTP1WhenUsingProxy(t *testing.T) {
+	c := configmock.New(t)
+
+	// Force client to use ALNP
+	c.SetWithoutSource("logs_config.http_protocol", "auto")
+	c.SetWithoutSource("skip_ssl_validation", true)
+
+	// The test server uses TLS, so if we set the http proxy (not https), it will still make
```
Comment (jhgilbert) [inline pkg/logs/client/http/destination_test.go:505]:
```suggestion
	// The test server uses TLS, so if we set the http proxy (not https), it still makes
```

**Review by jhgilbert — APPROVED** [review]
Approved with a very minor suggestion for style, thank you!

File: pkg/logs/client/http/destination_test.go [hunk 2]
```diff
@@ -494,3 +494,34 @@ func TestTransportProtocol_HTTP1FallBack(t *testing.T) {
 	// Assert that the server automatically falls back to HTTP/1.1
 	assert.Equal(t, "HTTP/1.1", resp.Proto)
 }
+
+func TestTransportProtocol_HTTP1WhenUsingProxy(t *testing.T) {
+	c := configmock.New(t)
+
+	// Force client to use ALNP
+	c.SetWithoutSource("logs_config.http_protocol", "auto")
+	c.SetWithoutSource("skip_ssl_validation", true)
```
Comment (remeh) [inline pkg/logs/client/http/destination_test.go:503]: You're sure the `skip_ssl_validation` is necessary? (I didn't check, I just find it surprising)

**Review by remeh — APPROVED** [review]
LGTM!

File: pkg/logs/client/http/destination_test.go (hunk 2 above)
Comment (gh123man) [reply pkg/logs/client/http/destination_test.go:503]: Yeah the test server enforces TLS (I assume somewhere under the hood with an unverified certificate). TLS is required to validate HTTP/2, so we have to disable validation for the test to work.

**Pull request merged**